### DISCUSSION TITLE
Modifying Job stores and dynamically adding/removing jobs

### DISCUSSION THREAD

**Discussion by gigwegbe** [agronholm/apscheduler#631] [Q&A]
Hello @agronholm,

Before using apscheduler, I had a MQTT client that subscribes to a topic and writes(adds/removes) the message to the SQLite database. The message payload contains the time and parameters(x, y, z) needed to act at a given time.

task_id | time | x | y | z
--- | --- | --- | --- |--- |
1 | 1:12 | 283 | 290 | 123
2 | 1:17 | 281 | 294 | 157

Now I want to use apscheduler to schedule tasks at an expected time and use the parameters(x,y,z) to call a function` def move(x,y,z)` and perform an action. Finally, I want the mqtt client to dynamically add/remove the schedules. 

First, question, how can I make the job stores have the same data format(task_id, time, x, y, z)?

Secondly question, most examples I have seen add a job as the script begins. Can I add a job dynamically from any messaging service like MQTT, firebase etc rather than add it at the beginning of the script? 
Thanks for apscheduler and I hope to hear from you soon.

**Comment by agronholm** [comment]
Please open a discussion next time you want to ask something. Issues are for bug reports and feature requests.

**Comment by agronholm** [comment]
> First, question, how can I make the job stores have the same data format(task_id, time, x, y, z)?

Why do you need to force the same data format on the stores? Why not just pass x, y and z as arguments to the scheduled job? For `time`, you can probably convert that to a timezone aware `datetime` object and use that with `DateTrigger`.

> Secondly question, most examples I have seen add a job as the script begins. Can I add a job dynamically from any messaging service like MQTT, firebase etc rather than add it at the beginning of the script?

Yes, you can add a job any time the scheduler is running.

**Comment by gigwegbe** [comment]
Thanks. I have implemented both dynamic adding/removing of tasks and mqtt integration. Also, how can I delete all jobs in the scheduler?
 ```
   job_list = scheduler.get_jobs()
    for job in job_list: 
        scheduler.remove_job(job)
```

**Reply by agronholm** [reply]
With this: https://apscheduler.readthedocs.io/en/3.x/modules/schedulers/base.html#apscheduler.schedulers.base.BaseScheduler.remove_all_jobs

**Comment by gigwegbe** [comment]
Thanks for the swift response. Apscheduler is awesome.
